pkg/destroy/powervs/serviceinstance.go

```diff
@@ -121,16 +121,24 @@ func (o *ClusterUninstaller) listServiceInstances() (cloudResources, error) {
 				o.Logger.Debugf("listServiceInstances: type: %v", *resourceInstance.Type)
 			}
 
-			if resourceInstance.Type != nil && *resourceInstance.Type == "service_instance" {
-				if strings.Contains(*resource.Name, o.InfraID) {
-					result = append(result, cloudResource{
-						key:      *resource.ID,
-						name:     *resource.Name,
-						status:   *resource.GUID,
-						typeName: serviceInstanceTypeName,
-						id:       *resource.ID,
-					})
-				}
+			if resourceInstance.Type == nil || resourceInstance.GUID == nil {
```

Comment:
Nit: This seems like to some common logic to determine if the resourceInstance is valid and of type "service instance". This should made onto a helper function that can be called from these 2 locations.